pydicom/charset.py

```diff
@@ -91,18 +91,38 @@ def decode_string(value, encodings):
                                             escape_codes['iso_ir_58']))
 
     if use_python_handling:
-        parts = [value]
+        values = [value]
     else:
+        # We handle only the delimiters actually appearing in the value
+        # to avoid too much overhead.
+        delims = []
```

Comment:
`delims = [dd for dd in delimiters if dd in value]`

Reply:
Yes, of course - but I'm removing this anyway now.